extensions/Khronos/KHR_draco_mesh_compression/README.md

```diff
@@ -0,0 +1,141 @@
+# KHR_draco_mesh_compression
+
+## Contributors
+
+* Fan Zhang, Google, <mailto:zhafang@google.com>
+* Ondrej Stava, Google, <mailto:ostava@google.com>
+* Frank Galligan, Google, <mailto:fgalligan@google.com>
+* Kai Ninomiya, Google, <mailto:kainino@google.com>
+* Patrick Cozzi, Cesium, [@pjcozzi](https://twitter.com/pjcozzi)
+
+## Status
+
+Draft
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension defines a schema to use Draco geometry compression libraries in glTF format. This allows glTF to support streaming compressed geometry data instead of the raw data.
+
+The [conformance](#conformance) section specifies what an implementation must do when encountering this extension, and how the extension interacts with the attributes defined in the base specification.
+
+## glTF Schema Updates
+
+Draco geometry compression library could be used for `primitive` by adding an `extension` property to a primitive, and defining its `KHR_draco_mesh_compression` property.
+
+The following picture shows the structure of the schema update. 
+
+**Figure 1**: Structure of geometry compression extension.
+![](figures/structure.png)
+
+In general, we will use the extension to point to the buffer that contains the compressed data. The major change is that the `accessor` in an extended `primitive` no
+longer points to a `bufferView`. Instead, the `attributes` of a primitive will use the decompressed data. This is valid because in glTF 2.0, `bufferView` is not required in `accessor`, although if it is not present, it will be used with `sparse` field to act as a sparse accessor. In this extension, we will ignore the `bufferView` property.
+
+Usage of the extension must be listed in the `extensionUsed` and `extensionsRequired`. 
+
+```javascript
+"extensionsUsed" : [
+    "KHR_draco_mesh_compression"
+]
+
+"extensionsRequired" : [
+    "KHR_draco_mesh_compression"
+]
+
+```
+
+The extension then could be used like the following, note that all other nodes stay the same
+except `primitives`:
+
+```javascript
+
+"mesh" : {
+    "primitives" : [
+        {
+            "attributes" : {
+                "POSITION" : 11,
+                "NORMAL" : 12,
+                "TEXCOORD_0" : 13,
+            },
+            "indices" : 10,
+            "mode" : 4
+            "extensions" : {
+                "KHR_draco_mesh_compression" : {
+                    "bufferView" : 5,
+                    "attributesOrder" : [
+                        "POSITION",
+                        "NORMAL",
+                        "TEXCOORD_0"
+                    ],
```

Comment:
The way you showed pushing attributes to accessors definitely works if an engine could do that. The problem we are considering is that in this way the primitives won't have any `accessors`, or the data decoded from extension would overwrite the existing `accessors`. Our purpose was to propose the extension in the way that it has minimum changes to the existing spec. That's why we want to preserve `accessors` even when the extension exists.
There might be some cases where overwriting `accessors` will cause issues. E.g.
1) Some attributes of the primitive are not compressed in the extension, normal `accessors` are used to store the attributes.
2) If some other extensions extended the way `accessors` are used for primitive attributes.
So to avoid such cases, we use the `attributesOrder` to refer to the `accessors` of the primitive so that the extension will have minimum impact to the spec even with more complicated cases in the future.